### ISSUE TITLE
Async/Generator function lengths can't be configured

### ISSUE THREAD

**Issue by rhuanjl** [Microsoft/ChakraCore#5419]
Issue I noticed whilst working on #5405 async/generator functions report that their length properties are configurable but attempting to delete them or redefine with Object.defineProperty silently fails.

I intend to submit a PR to fix this after #5405 is merged - issue created as a reminder/in case #5405 can't be merged.

Example:
```js
function* gen(a, b) { }
async function asy(a, b, c) { }
print ("generator function.length property exists: " + gen.hasOwnProperty("length"));
print ("async function.length property exists: " + asy.hasOwnProperty("length"));

print("async function.length configurable = " + Object.getOwnPropertyDescriptor(asy, "length").configurable);
print("generator function.length configurable = " + Object.getOwnPropertyDescriptor(gen, "length").configurable);

print ("async function property names: " + Object.getOwnPropertyNames(asy));
print ("generator function property names: " + Object.getOwnPropertyNames(gen));

delete gen.length
if (gen.length !== 0) {
    print("Deleting generator function length failed");
}

delete asy.length;
if (asy.length !== 0) {
    print("Deleting async function length failed");
}
```

Eshost output:
```
#### V8
generator function.length property exists: true
async function.length property exists: true
async function.length configurable = true
generator function.length configurable = true
async function property names: length,name
generator function property names: length,name,prototype

#### Chakra
generator function.length property exists: true
async function.length property exists: true
async function.length configurable = true
generator function.length configurable = true
async function property names: name
generator function property names: prototype,name
Deleting generator function length failed
Deleting async function length failed
```

Starting point - Delete property handler for Generator function BUT note the type handler for a generator function is not given a length property so just changing this point wouldn't be enough:
https://github.com/Microsoft/ChakraCore/blob/94434008819dfed6e8ede678526875a67cfaa614/lib/Runtime/Library/JavascriptGeneratorFunction.cpp#L483-L489

EDIT: note these lengths also don't get returned by Object.getOwnPropertyNames

**Comment by kfarnung** [comment]
Thanks @rhuanjl!

**Comment by rhuanjl** [comment]
EDIT: see above added an additional related error.

Note: the fix I have ready to submit fixes the additional error too. (BUT I can't yet submit the fix as it is dependent on #5405 )
